### ISSUE TITLE
Wildcard SAN issue

### ISSUE THREAD

**Issue by jangrewe** [silkeh/pdns_api.sh#16]
Just a heads up, there seems to be an issue with certificates that contain the wildcard name as a SAN.

First reported issue: https://github.com/lukas2511/dehydrated/issues/420#issuecomment-372954837
More detailed issue: https://github.com/lukas2511/dehydrated/issues/504

**Comment by rhaamo** [comment]
If it helps, with debugging this is how pdns_api talk to pdns:
```
Name:  _acme-challenge.bbs.foobar.tld.
Token: VALUE1
Zone:  foobar.tld.
Name:  _acme-challenge.lan.foobar.tld.
Token: VALUE2
Zone:  foobar.tld.
Name:  _acme-challenge.lan.foobar.tld.
Token: VALUE3
Zone:  foobar.tld.
Method: PATCH
URL: http://10.0.0.201:8000/api/v1/servers/localhost/zones/foobar.tld.
Data: {"rrsets": [{
    "name": "_acme-challenge.bbs.foobar.tld.",
    "type": "TXT",
    "ttl": 1,
    "records": [{
      "content": "\"VALUE1\"",
      "disabled": false,
      "set-ptr": false
      
    }],
    "changetype": "REPLACE"
  },{
    "name": "_acme-challenge.lan.foobar.tld.",
    "type": "TXT",
    "ttl": 1,
    "records": [{
      "content": "\"VALUE2\"",
      "disabled": false,
      "set-ptr": false
      
    }],
    "changetype": "REPLACE"
  },{
    "name": "_acme-challenge.lan.foobar.tld.",
    "type": "TXT",
    "ttl": 1,
    "records": [{
      "content": "\"VALUE3\"",
      "disabled": false,
      "set-ptr": false
      
    }],
    "changetype": "REPLACE"
  }]}
Response: 
Method: PUT
URL: http://10.0.0.201:8000/api/v1/servers/localhost/zones/foobar.tld./notify
Data: 
Response: {"result": "Notification queued"}
```

In the end after that PATCH and PUT I can find only two records instead of three.

**Comment by silkeh** [comment]
The RRset replaces two records in the same place. What the hook probably *should* do is:

```
Data: {"rrsets": [{
    "name": "_acme-challenge.bbs.foobar.tld.",
    "type": "TXT",
    "ttl": 1,
    "records": [{
      "content": "\"VALUE1\"",
      "disabled": false,
      "set-ptr": false
    }],
    "changetype": "REPLACE"
  },{
    "name": "_acme-challenge.lan.foobar.tld.",
    "type": "TXT",
    "ttl": 1,
    "records": [{
      "content": "\"VALUE2\"",
      "disabled": false,
      "set-ptr": false
    },{
      "content": "\"VALUE3\"",
      "disabled": false,
      "set-ptr": false
    }],
    "changetype": "REPLACE"
  }]}
```

It should probably work without `HOOK_CHAIN`.

**Comment by jangrewe** [comment]
@qvr's fix in #17 works fine for me!

**Comment by silkeh** [comment]
Reopening this for visibility. It will be closed once #17 goes in.
